src/core/hle/kernel/k_thread.cpp

```diff
@@ -308,14 +308,20 @@ void KThread::Finalize() {
 
         auto it = waiter_list.begin();
         while (it != waiter_list.end()) {
-            // Clear the lock owner
-            it->SetLockOwner(nullptr);
+            // Get the thread.
+            KThread* const waiter = std::addressof(*it);
```

Comment:
```suggestion
            KThread& waiter = *it;
```

Seems much nicer, no?

Reply:
This was chosen intentionally to [match the code from Atmosphere](https://github.com/Atmosphere-NX/Atmosphere/blob/e702eab21c1ecc25015f246480e29d2d60f457eb/libraries/libmesosphere/source/kern_k_thread.cpp#L403-L419). I have noticed that even small differences from the reference implementation can cause significant issues, so I would prefer to leave it as is in this case.